### PR TITLE
Fetching Cumulative visits from GA4, reading UA report from GCP buckets

### PR DESCRIPTION
**Pull request by orzut**

### CODE REVIEW

File: src/pkg/caendr/caendr/services/cloud/analytics.py [hunk 1]
```diff
@@ -37,6 +45,7 @@ def get_weekly_visits(SERVICE):
       'reportRequests': [{
         'viewId': '117392266',
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/analytics.py]: not part of the PR, but avoid having `magic numbers`, try to create constants and reference them, or consider moving these to part of the config (ENV VARS) for the app. 
E.g.: GOOGLE_ANALYTICS_VIEW_ID: 117392266

File: src/pkg/caendr/caendr/services/cloud/analytics.py [hunk 2]
```diff
@@ -37,6 +45,7 @@ def get_weekly_visits(SERVICE):
...
         'dateRanges': [{'startDate': '2015-01-01', 'endDate': datetime.now().date().isoformat()}],
+        # 'dateRanges': [{'startDate': '2015-01-01', 'endDate': '2023-03-31'}],
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/analytics.py]: remove commented code.

File: src/pkg/caendr/caendr/services/cloud/analytics.py [hunk 3]
```diff
@@ -37,6 +45,7 @@ def get_weekly_visits(SERVICE):
       'reportRequests': [{
         'viewId': '117392266',
         'dateRanges': [{'startDate': '2015-01-01', 'endDate': datetime.now().date().isoformat()}],
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/analytics.py]: not part of the PR, but consider creating a constant for `2015-01-01` that offers semantic meaning (aligned with the existing comment).

File: src/pkg/caendr/caendr/services/cloud/analytics.py [hunk 4]
```diff
@@ -52,6 +61,9 @@ def get_weekly_visits(SERVICE):
   df = pd.DataFrame(out) \
           .sort_values('date') \
           .reindex(['date', 'count'], axis=1)
+  
+  # df.to_csv('archived_ga_data.csv', index=False)
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/analytics.py]: remove commented out code

File: src/pkg/caendr/caendr/services/cloud/google_analytics_4.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  service_credentials = get_service_account_credentials(get_secret(GOOGLE_ANALYTICS_SERVICE_ACCOUNT_NAME))
+  scope = ['https://www.googleapis.com/auth/analytics.readonly']
+  return ServiceAccountCredentials.from_json_keyfile_dict(service_credentials, scope)
+
+
+@use_service('analyticsdata', 'v1beta', credentials=get_analytics_credentials)
+def get_weekly_visits_ga4(SERVICE):
+  """ Get reports from Google Analytics 4 """
+
+  property_id = '361534565'
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/google_analytics_4.py]: consider moving the property_id to the config of the app (ENV VAR).

File: src/pkg/caendr/caendr/services/cloud/google_analytics_4.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  property = f"properties/{property_id}"
+  request = {
+    "dateRanges": [
+    {
+        "startDate": "2023-04-01",
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/google_analytics_4.py]: consider creating a constant here with the `startDate`, eg: GOOGLE_ANALYTICS_V4_START_DATE = "2023-04-01"`.

File: src/pkg/caendr/caendr/services/cloud/google_analytics_4.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+    *** UA had been tracking data for CaeNDR before 2023/03/31
+        Starting 2023/04/01 it was migrated to GA4.
+        UA stopped proccessing data starting 2023/07/01.
+  """
+  
+  MODULE_SITE_BUCKET_PRIVATE_NAME = os.environ.get('MODULE_SITE_BUCKET_PRIVATE_NAME')
+
+  # Get the CSV file from buckets
+  blob = get_blob(MODULE_SITE_BUCKET_PRIVATE_NAME, 'ga_reports/archived_ga_data_for_plot.csv')
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/google_analytics_4.py:72]: 👍

File: src/pkg/caendr/caendr/services/cloud/google_analytics_4.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    "dimensions": [{'name': 'year'}, {'name': 'week'}],
+    "metrics": [{'name': 'sessions'}],
+    'orderBys': {
+      'desc': True,
+      'metric': {
+        'metricName': 'sessions'
+      }
+    }
+  }
+  response = SERVICE.properties().runReport(property=property, body=request).execute()
```
Comment (r-vieira) [inline src/pkg/caendr/caendr/services/cloud/google_analytics_4.py:50]: Are we able to fail gracefully if the GoogleAnalytics service is down?

**Review by r-vieira — CHANGES_REQUESTED** [review]
looks good! Please see inline comments

File: src/pkg/caendr/caendr/services/cloud/analytics.py (hunk 2 above)
Comment (orzut) [reply src/pkg/caendr/caendr/services/cloud/analytics.py]: removed in the commit [4e1e299](https://github.com/AndersenLab/CAENDR/pull/408/commits/4e1e29981e3dad85cd0ec00ef2fbd2610ad5f949)

File: src/pkg/caendr/caendr/services/cloud/analytics.py (hunk 4 above)
Comment (orzut) [reply src/pkg/caendr/caendr/services/cloud/analytics.py]: removed in the commit [4e1e299](https://github.com/AndersenLab/CAENDR/pull/408/commits/4e1e29981e3dad85cd0ec00ef2fbd2610ad5f949)

File: src/pkg/caendr/caendr/services/cloud/google_analytics_4.py (hunk 5 above)
Comment (orzut) [reply src/pkg/caendr/caendr/services/cloud/google_analytics_4.py]: fixed in the commit [ac2f630](https://github.com/AndersenLab/CAENDR/pull/408/commits/ac2f6309f446117b209c29df9d25ebfc661f045e)

File: src/pkg/caendr/caendr/services/cloud/google_analytics_4.py (hunk 6 above)
Comment (orzut) [reply src/pkg/caendr/caendr/services/cloud/google_analytics_4.py]: fixed in the commit [ac2f630](https://github.com/AndersenLab/CAENDR/pull/408/commits/ac2f6309f446117b209c29df9d25ebfc661f045e)

File: src/pkg/caendr/caendr/services/cloud/google_analytics_4.py (hunk 8 above)
Comment (orzut) [reply src/pkg/caendr/caendr/services/cloud/google_analytics_4.py:50]: We're handling it with `use_service()` function in the `src/pkg/caendr/caendr/services/cloud/discovery.py` file
